nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -413,7 +413,7 @@ def criterion_patch(self, criterion: Callable[[Tensor, Tensor], Tensor]) -> Call
         def patched_criterion(input_tensor: Tensor, target: Tensor):
             sum_l1 = 0
             for _, wrapper in self.get_modules_wrapper().items():
-                sum_l1 += torch.norm(wrapper.module.weight.data, p=1)
+                sum_l1 += torch.norm(wrapper.module.weight, p=1)
```

Comment:
Does this matter final result?
I ran the experiment with SlimPruner in two ways (use `.data` or not) and found almost no difference : (

Reply:
what is the difference between "weight.data" and "weight"?

Reply:
![image](https://user-images.githubusercontent.com/33053116/142790271-3632e9b3-36d7-43b2-bcd2-09e0b4cb95db.png)

`weight` has `grad_fn` but `weight.data` hasn't.